test/Token.js

```diff
@@ -62,30 +61,29 @@ describe("Token contract", function () {
     });
 
     it("Should assign the total supply of tokens to the owner", async function () {
+      const { hardhatToken, owner } = await loadFixture(deployToken);
       const ownerBalance = await hardhatToken.balanceOf(owner.address);
       expect(await hardhatToken.totalSupply()).to.equal(ownerBalance);
     });
   });
 
   describe("Transactions", function () {
     it("Should transfer tokens between accounts", async function () {
+      const { hardhatToken, owner, addr1, addr2 } = await loadFixture(deployToken);
       // Transfer 50 tokens from owner to addr1
-      await hardhatToken.transfer(addr1.address, 50);
-      const addr1Balance = await hardhatToken.balanceOf(
-        addr1.address
-      );
-      expect(addr1Balance).to.equal(50);
+      await expect(hardhatToken.transfer(addr1.address, 50))
+        .to.emit(hardhatToken, "Transfer").withArgs(addr1, 50)
+        .and.to.changeTokenBalances(hardhatToken, [owner, addr1], [-50, 50]);
 
       // Transfer 50 tokens from addr1 to addr2
       // We use .connect(signer) to send a transaction from another account
-      await hardhatToken.connect(addr1).transfer(addr2.address, 50);
-      const addr2Balance = await hardhatToken.balanceOf(
-        addr2.address
-      );
-      expect(addr2Balance).to.equal(50);
+      await expect(hardhatToken.connect(addr1).transfer(addr2.address, 50))
+        .to.emit(hardhatToken, "Transfer").withArgs(addr2, 50)
```

Comment:
I think it's better if we remove the `to.emit` calls from this test, and create a new one exclusively for the events.

`    it("Should emit the Transfer event", async function () {`

Reply:
Addressed in https://github.com/NomicFoundation/hardhat-boilerplate/pull/61/commits/e20e7159222834378fd30a309264424b91634c98